### DISCUSSION TITLE
regarding PartialPathGraph

### DISCUSSION THREAD

**Discussion by ice-doom** [github/codeql#7555] [Q&A] [answered]
myself ql rule:
![image](https://user-images.githubusercontent.com/25363717/148872509-a4d038c2-894d-4971-a63a-9367f83620d7.png)
`conf.hasPartialFlow` did not get the correct results, it stay at `getHost()`
![image](https://user-images.githubusercontent.com/25363717/148872581-a835c122-17dc-4046-a640-9c615c92c2df.png)

when use `PathGraph` `conf.hasFlowPath` The correct results can be obtained
![image](https://user-images.githubusercontent.com/25363717/148872696-99cda2bd-8b10-4605-8263-0ee67c06e5a2.png)

What caused it

**Comment by smowton** [comment, marked as answer]
That truncated partial path looks right to me: the function `getHost` returns a string constant, so it is correct that there is taint in the `vul_name` field of `this`, but that the `getHost` function does not return it.

You will surely find that one of the other partial paths proposed recreates the full flow to `rt.exec` that you see with a normal `PathGraph`.

**Comment by ice-doom** [comment]
I found what I wanted, but the results are in random order. If the amount of code is large and there are thousands of results, how can I use `conf.hasFlowPath` to filter them out?

**Reply by smowton** [reply]
I often find the easiest way is to restrict the source-location of the sink -- "I'm interested in partial paths that end in file Xyz.java lines 101-104" sort of thing. Note for partial paths the "sink" is the end of the partial path, not the sink as defined in the Configuration.
